tests/types/test_uuid.py

```diff
@@ -32,7 +41,13 @@ def test_uuid_load(conn, fmt_out):
         copy.set_types(["uuid"])
         (res,) = copy.read_row()
 
-    assert res == UUID(val)
+    uuid_val = UUID(val)
+    assert res == uuid_val
+    # the C modules bypasses __init__, so checking the state of the UUID object
+    assert res.hex == uuid_val.hex
+    assert res.int == uuid_val.int
+    assert res.bytes == uuid_val.bytes
+    assert res.is_safe == uuid_val.is_safe
 
 
 @pytest.mark.slow
```

Comment:
I would like to add a test to verify that the UUID structure doesn't change in future versions; if it does we might need to adapt our loaders.

```python
assert uuid.UUID.__slots__ == ("int", "is_safe", "__weakref__"), "UUID structure changed"
```

Reply:
Added!